pandasai/smart_datalake/generate_smart_datalake_pipeline.py

```diff
@@ -0,0 +1,31 @@
+from typing import Optional
+from pandasai.helpers.logger import Logger
+from pandasai.pipelines.pipeline import Pipeline
+from pandasai.pipelines.pipeline_context import PipelineContext
+from pandasai.smart_datalake.code_execution import CodeExecution
+from pandasai.smart_datalake.code_generator import CodeGenerator
+from pandasai.smart_datalake.result_parsing import ResultParsing
+from pandasai.smart_datalake.result_validation import ResultValidation
+
+
+class GenerateSmartDatalakePipeline:
+    _pipeline: Pipeline
+
+    def __init__(
+        self,
+        context: Optional[PipelineContext] = None,
+        logger: Optional[Logger] = None,
+    ):
+        self._pipeline = Pipeline(
+            context=context,
+            logger=logger,
+            steps=[
+                CodeGenerator(),
```

Comment:
@milind-sinaptik can we add one more logic unit before code generator for generate prompt and use one that is already there for run execute llm? You can check the example of already implemented pipeline of sythetic dataframe.

Reply:
done